src/input.rs

```diff
@@ -84,9 +87,17 @@ impl<'a> InputKind<'a> {
     }
 }
 
+#[derive(Clone)]
+pub(crate) struct InputPermissions {
+    pub(crate) mode: u32,
```

Comment:
Apparently getting user and group is not supported in the std library and we would need to develop further for that, so I left that as a later improvement to avoid complicating this PR further.